.github/workflows/matrix-tests.yaml

```diff
@@ -6,14 +6,29 @@ on:
   workflow_dispatch:
 
 jobs:
-  main:
+  set-matrix:
     runs-on: ubuntu-latest
-    name: Run the interface tests on all registered charms
+    outputs:
+      matrix_values: ${{ steps.set-matrix.outputs.matrix_values }}
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
         with:
           fetch-depth: 1
+      - name: Set matrix values
+        id: set-matrix
+        run: |
+          interfaces=[$(find interfaces -mindepth 1 -maxdepth 1 -type d -not -name "__template__" -printf '"%f",')]
+          echo "matrix_values=${interfaces}" >> $GITHUB_OUTPUT
+
+  main:
+    name: Run the interface tests on all registered charms
```

Comment:
We should definitely prefix this with `${{ matrix.interface }}`. Maybe we can just make the name the interface name, as that will be much easier to find stuff in Actions: https://github.com/IronCore864/charm-relation-interfaces/actions/runs/11790190571/job/32840217966

Reply:
Job name updated.